### ISSUE TITLE
fisheye calibrate requires all images to have the same number of calibration points

### ISSUE THREAD

**Issue by mcbridejc** [opencv/opencv#6388]
It appears that fisheye::calibrate will sometimes fail with the error below (depending on the ordering of images provided) when not all provided images have the same number of calibration points. 

```
cv2.error: /home/vagrant/opencv/modules/core/src/matrix.cpp:456: error: (-215) 0 <= _rowRange.start && _rowRange.start <= _rowRange.end && _rowRange.end <= m.rows in function Mat
```

After some debugging, I found that this error is occuring in cv::internal::EstimateUncertainties in modules/calib3d/src/fisheye.cpp, because the number of points in the first objectPoints matrix is used to compute the total number of points for all images: 

```
Mat ex((int)(objectPoints.getMat(0).total() * objectPoints.total()), 1, CV_64FC2);

    for (int image_idx = 0; image_idx < (int)objectPoints.total(); ++image_idx)
    {
        Mat image, object;
        objectPoints.getMat(image_idx).convertTo(object, CV_64FC3);
        imagePoints.getMat (image_idx).convertTo(image, CV_64FC2);

        Mat om(omc.getMat().col(image_idx)), T(Tc.getMat().col(image_idx));

        std::vector<Point2d> x;
        projectPoints(object, x, om, T, params, noArray());
        Mat ex_ = image.t() - Mat(x);
        printf("copyTo range (%d, %d)", ex_.rows * image_idx, ex_.rows * (image_idx + 1));
        ex_.copyTo(ex.rowRange(ex_.rows * image_idx,  ex_.rows * (image_idx + 1)));
    }
```

This issue appears to be present in 3.0.0, as well as 3.1.0. 

I suppose one could argue that this is a feature request not a bug, however, the cv::calibrateCamera does not impose this constraint. Also, I don't believe there's any reason that the algorithm should need to have the same set of points in each image, and in fact as far as I can see the only place it causes an issue is while computing the error. Happy to hear if I've misunderstood something here. If having the same number of points is a requirement, then I'd recommend adding a note to the documentation and an assert statement to make it more clear why the function fails when it does.

## Reproduction 

This python code reproduces the issue for me:

```
import cv2
import numpy as np

NGRID = 10

# Create a grid of 10x10 array of points, and a grid of 10x11
# The same points will be used for image and object points 
range1 = np.arange(-NGRID, NGRID, dtype=np.float32)
range2 = np.arange(-NGRID, NGRID+1, dtype=np.float32)
imgPoints = [
    np.dstack(np.meshgrid(range1, range1, dtype=np.float32)).reshape(-1, 2),
    np.dstack(np.meshgrid(range1, range2, dtype=np.float32)).reshape(-1, 2)
]

objPoints = [np.hstack([x, np.zeros((len(x), 1), dtype=np.float32)]) for x in imgPoints]

# Setup input arguments to appropriate shapes for fisheye::calibrate
# This was based on https://github.com/Itseez/opencv/issues/5534
cam_mtx = np.zeros((3, 3), dtype=np.float32)
cam_mtx[0, 0] = 1.0
cam_mtx[1, 1] = 1.0
cam_mtx[0, 2] = 0.0
cam_mtx[1, 2] = 0.0
cam_mtx[2, 2] = 1.0
dist = np.zeros((4, 1), dtype=np.float64)

# Re-size for opencv type assertions to be happy. This was hard-earned.
objPoints = [np.transpose(np.resize(x, (len(x), 1, 3)), axes=[1, 0, 2]) for x in objPoints]
imgPoints = [np.transpose(np.resize(x, (len(x), 1, 2)), axes=[1, 0, 2]) for x in imgPoints]

rvecs = [np.zeros((1, 1, 3), dtype=np.float64) for _ in range(len(objPoints))]
tvecs = [np.zeros((1, 1, 3), dtype=np.float64) for _ in range(len(objPoints))]

ret, dummy, dummy, dummy, dummy = cv2.fisheye.calibrate(
    objPoints,
    imgPoints,
    (1280, 970),
    cam_mtx,
    dist,
    rvecs,
    tvecs,
    flags=cv2.fisheye.CALIB_USE_INTRINSIC_GUESS | cv2.fisheye.CALIB_RECOMPUTE_EXTRINSIC | cv2.fisheye.CALIB_FIX_SKEW)

print "Camera Matrix: " + str(cam_mtx)
print "Dist coeffs: "  + str(dist)
print "Reprojection error: " + str(ret)
```

**Comment by mcbridejc** [comment]
I made the changes shown in pull request #6396 above and now I'm able to use fisheye::calibrate() successfully.

**Comment by sovrasov** [comment]
resolved in #7026
